### PR TITLE
get mem on cgroupsv2

### PR DESCRIPTION
**Pull request by KannarFr**
Description of your changes:

Fixes: #2118

### CODE REVIEW

**Comment by KannarFr** [comment]
I checked the failing test but got

```


Identified problems
--
  | No identified problemNo problems were identified. If you know why this problem occurred, please add a suitable Cause for it.
```

Shall I run a bash linter or something?

**Comment by guits** [comment]
jenkins test sloppy bash

**Comment by guits** [comment]
jenkins lint

**Comment by guits** [comment]
```

In /common_functions.sh line 525:
  if grep -q cgroup2 /proc/filesystems; then
  ^-- SC1009 (info): The mentioned syntax error was in this if expression.


In /common_functions.sh line 529:
  elif
  ^-- SC1049 (error): Did you forget the 'then' for this 'elif'?
  ^-- SC1073 (error): Couldn't parse this elif clause. Fix to allow more checks.


In /common_functions.sh line 533:
  fi
    ^-- SC1072 (error): Unexpected keyword/token. Fix any mentioned problems and try again.
```

File: src/daemon/common_functions.sh [hunk 1]
```diff
@@ -519,16 +519,28 @@ function ami_privileged {
...
 function get_available_ram {
-  limit_in_bytes="/sys/fs/cgroup/memory/memory.limit_in_bytes"
-  memory_limit=$(cat $limit_in_bytes)
+  memory_limit_in_bytes_file=""
+  memory_current_usage_in_bytes_file=""
+
+  if grep -q cgroup2 /proc/filesystems; then
+    # cgroups v2
+    memory_limit_in_bytes_file="/sys/fs/cgroup/memory.max"
+    memory_current_usage_in_bytes_file="/sys/fs/cgroup/memory.current"
```
Comment (guits) [inline src/daemon/common_functions.sh:528]: @KannarFr 

```console
[root@7f9056067200 /]# grep cgroup2 /proc/filesystems
nodev	cgroup2
[root@7f9056067200 /]# ls /sys/fs/cgroup/memory.max
ls: cannot access '/sys/fs/cgroup/memory.max': No such file or directory
[root@7f9056067200 /]# ls /sys/fs/cgroup/memory.current
ls: cannot access '/sys/fs/cgroup/memory.current': No such file or directory
[root@7f9056067200 /]#
```

from the documentation [1] :
>  memory.usage_in_bytes		 # show current usage for memory
 memory.limit_in_bytes		 # set/show limit of memory usage


did you mean `/sys/fs/cgroup/memory/memory.limit_in_bytes`  `/sys/fs/cgroup/memory/memory.usage_in_bytes` ?

**Review by guits — CHANGES_REQUESTED** [review]

File: src/daemon/common_functions.sh [hunk 2]
```diff
@@ -519,16 +519,28 @@ function ami_privileged {
 
 # Detect how much ram is available
 function get_available_ram {
-  limit_in_bytes="/sys/fs/cgroup/memory/memory.limit_in_bytes"
-  memory_limit=$(cat $limit_in_bytes)
+  memory_limit_in_bytes_file=""
+  memory_current_usage_in_bytes_file=""
+
+  if grep -q cgroup2 /proc/filesystems; then
```
Comment (guits) [inline src/daemon/common_functions.sh:525]: shouldn't it be checked from `/proc/mounts` instead?

```console
# grep cgroup2 /proc/mounts
cgroup2 /sys/fs/cgroup cgroup2 rw,seclabel,nosuid,nodev,noexec,relatime,nsdelegate 0 0
#
```

File: src/daemon/common_functions.sh (hunk 1 above)
Comment (guits) [reply src/daemon/common_functions.sh:528]: @KannarFr forget about this comment, I think I got confused.

File: src/daemon/common_functions.sh [hunk 3]
```diff
@@ -519,16 +519,28 @@ function ami_privileged {
...
   # 8 ExaBytes is the value of an unbounded device
-  if  [ "${memory_limit}" = "9223372036854771712" ]; then
+  if [ "${memory_limit}" = "9223372036854771712" ] || [ "${memory_limit}" = "max" ]; then
     # Looks like we are not in a container
     # Let's report the MemAvailable on this system
     echo $(( $(awk '/MemAvailable/{print $2}' /proc/meminfo) * 1024))
   else
-    current_usage=$(cat /sys/fs/cgroup/memory/memory.usage_in_bytes)
-    echo $(( memory_limit - current_usage ))
+    memory_current_usage=$(cat memory_current_usage_in_bytes_file)
```
Comment (guits) [inline src/daemon/common_functions.sh:542]:
```suggestion
    memory_current_usage=$(cat $memory_current_usage_in_bytes_file)
```

**Review by guits — CHANGES_REQUESTED** [review]

**Review by guits — COMMENTED** [review]
not related to this patch but if you want to address the warning `SC2071` L124 you can apply this small change:

```diff
diff --git a/src/daemon/common_functions.sh b/src/daemon/common_functions.sh
index 1b679c6c..9e99046f 100755
--- a/src/daemon/common_functions.sh
+++ b/src/daemon/common_functions.sh
@@ -121,7 +121,7 @@ function dev_part {
       if [[ $(readlink -f "$option") == "$desired_partition" ]]; then
         local optprefixlen
         optprefixlen=$(prefix_length "$option" "$osd_device")
-        if [[ $optprefixlen > $pfxlen ]]; then
+        if [[ $optprefixlen -gt $pfxlen ]]; then
           link=$option
           pfxlen=$optprefixlen
         fi
```

**Review by guits — CHANGES_REQUESTED** [review]
@KannarFr you need to add a sign off to your commit message

thanks!

**Comment by KannarFr** [comment]
@guits thanks for the hints. I applied all of them. I also sign off it.

**Comment by guits** [comment]
@KannarFr Could you squash the first 3 commits? I think you should have only 2 commits in this PR.

`get mem on cgroupsv2` and `apply unrelated lint fix`

Thanks!

**Comment by KannarFr** [comment]
@guits Damn I thought github will do it by itself. Done.

**Review by guits — APPROVED** [review]

**Comment by guits** [comment]
failure in `container / arm64` is unrelated to this patch

**Comment by KannarFr** [comment]
@guits can you make a release with this patch? :)

**Comment by guits** [comment]
@KannarFr which release are you looking for?

**Comment by KannarFr** [comment]
`ceph/daemon`

**Pull request merged**